views/js/e2e/runner/_setup/setupCommands.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+import setupData from './setupData';
+import setupSelectors from './setupSelectors';
+import base64Test from './base64QtiExampleTestPackage';
+
+/**
+ * Setup Commands
+ */
+Cypress.Commands.add('publishImportedTest', () => {
+    
+    // Register route for test deletion
+    cy.route('POST', '**/editTest').as('editTest');
+    
+    // Visit Tests page
+    cy.visit(setupData.testsPageUrl);
+    
+    // Select test import
+    cy.get(setupSelectors.testsPage.testImportbutton).click();
+
+    // Upload example qti test file to file input
+    cy.get(setupSelectors.testsPage.fileInput).upload(
+        {
+            fileContent: base64Test, 
+            fileName: 'e2eExampleTest.zip', 
+            mimeType: 'application/zip'
+        }, 
+        { 
+            subjectType: 'input' 
+        }
+    );
+
+    // Import selected example test file
+    cy.get(setupSelectors.testsPage.fileImportButton).click();
+
+    // Continue
+    cy.get(setupSelectors.testsPage.feedbackContinueButton).click();
+
+    // Wait until publish button appears again
+    cy.wait('@editTest');
+
+    // Publish example test
+    cy.get(setupSelectors.testsPage.testPublishButton).click();
+
+    // Select Assembled Delivery as root class for publishing
+    cy.get(setupSelectors.testsPage.destinationSelector).contains('Assembled Delivery').click();
+
+    // Clicking on publish
+    cy.get(setupSelectors.testsPage.destinationSelectorActions).contains('Publish').click();
+});
+
+Cypress.Commands.add('setDeliveryForGuests', () => {
```

Comment:
The command could be useful in future for any delivery, not just this one. I guess the name or id could be a parameter.

Reply:
Not for the testrunner atm. But you are right, maybe all such cleanup/setup commands could be abstract and could go to the tao-e2e-runner but we don't know yet for what will be used for in the future. Currently there is no plan to add more e2e tests after these runner related are finished. I'd say we can discuss afterwards.